caching/src/main/java/com/iluwatar/caching/App.java

```diff
@@ -1,58 +1,53 @@
+ * which writes data into the cache initially whilst the data is only
+ * written into the DB when the cache is full, and <code>cache-aside</code>
+ * which pushes the responsibility of keeping the data synchronized in both
+ * data sources to the application itself. The <code>read-through</code>
+ * strategy is also included in the mentioned four strategies --
+ * returns data from the cache to the caller <b>if</b> it exists <b>else</b>
+ * queries from DB and stores it into the cache for future use. These strategies
+ * determine when the data in the cache should be written back to the backing
+ * store (i.e. Database) and help keep both data sources
+ * synchronized/up-to-date. This pattern can improve performance and also helps
+ * to maintainconsistency between data held in the cache and the data in
+ * the underlying data store.
  *
- * <p>In this example, the user account ({@link UserAccount}) entity is used as the underlying
- * application data. The cache itself is implemented as an internal (Java) data structure. It adopts
- * a Least-Recently-Used (LRU) strategy for evicting data from itself when its full. The four
- * strategies are individually tested. The testing of the cache is restricted towards saving and
- * querying of user accounts from the underlying data store ( {@link DbManager}). The main class (
- * {@link App} is not aware of the underlying mechanics of the application (i.e. save and query) and
- * whether the data is coming from the cache or the DB (i.e. separation of concern). The AppManager
- * ({@link AppManager}) handles the transaction of data to-and-from the underlying data store
- * (depending on the preferred caching policy/strategy).
+ * <p>In this example, the user account ({@link UserAccount}) entity is used
+ * as the underlying application data. The cache itself is implemented as an
+ * internal (Java) data structure. It adopts a Least-Recently-Used (LRU)
+ * strategy for evicting data from itself when its full. The four
+ * strategies are individually tested. The testing of the cache is restricted
+ * towards saving and querying of user accounts from the
+ * underlying data store( {@link DbManager}). The main class ( {@link App}
+ * is not aware of the underlying mechanics of the application
+ * (i.e. save and query) and whether the data is coming from the cache or the
+ * DB (i.e. separation of concern). The AppManager ({@link AppManager}) handles
+ * the transaction of data to-and-from the underlying data store (depending on
+ * the preferred caching policy/strategy).
  * <p>
- * <i>{@literal App --> AppManager --> CacheStore/LRUCache/CachingPolicy --> DBManager} </i>
+ * <i>{@literal App --> AppManager --> CacheStore/LRUCache/CachingPolicy -->
+ * DBManager} </i>
+ * </p>
+ *
+ * <p>
+ * To run the application with MongoDb:
+ * 1. Launch mongoDB in docker container with command: docker-compose up
+ * 2. Start application with parameter --mongo
+ * Example: java -jar app.jar --mongo
```

Comment:
Thanks for providing the instructions. However, I think they should be more detailed.

- Please link to docker-compose installation instructions. What platforms have you tested on?
- Does the application always need to be started with --mongo parameter? What happens if it is left out?